Utils/Extract.pm

```diff
@@ -195,6 +195,7 @@ sub extract_file_to_temp_cache {
     my $filename = shift;
     my $suffix = shift;
     my $input_cache_dir = shift;
+    my $metrics = shift;
```

Comment:
I don't love having to pass through the metrics object down through all the functions here; I would rather be able to get them from some global context (like `Services`), or like we might with a logger; we don't have the sort-of-global context `$C` available here.

Reply:
`MdpGlobals`?

Reply:
Having looked at the Metrics.pm file in the imgsrv PR, it seems to me that the non-imgsrv-specific parts of that module should probably be relocated here if the likelihood is that a singleton, long-running metrics object would be appropriate. Then perhaps in the app's setup code we could have something like `$global_metrics->register("mdpitem_get_mdpitem_seconds", ...)` which would set a named Prometheus accessor. Leaf nodes in mdp-lib would report to this global hub without tramping context data thru the call chain. Leaf procedures that call `$global_metrics->observe("some_metric_this_app_doesnt_care_about", ...)` will just result in a no-op (or could be checked by the caller, e.g. `if($global_metrics->can_observe(...)) { ... }` if the metrics calculation is expensive).

I don't know how this might impact the Plack side, i.e., the observability part (`call`, `render`, `finalize`, maybe others) but for now I *think* that's imgsrv-specific and could be separated out as an add-on.

Reply:
I'll have to think about how to do that, and it might be worthwhile to look at this together.

Reply:
See comments on imgsrv PR -- we will get this from a singleton metrics thing like `HTFeed::JobMetrics`

Reply:
We would need to think about how to set up the metrics here -- one thing we could do is at the time of recording a metrics, check if the metrics have been set up & if not set them up...